modules/graderStake/s4grader.go

```diff
@@ -0,0 +1,61 @@
+package graderStake
+
+var _ BlockGraderV4 = (*S4BlockGrader)(nil)
+
+// S4BlockGrader implements the S4 grading algorithm.
+// Entries are encoded in Protobuf with 100 winners each block.
+// Valid assets can be found in ´opr.V5Assets´
+type S4BlockGrader struct {
+	baseGrader
+}
+
+// Version 8
+func (s4 *S4BlockGrader) Version() uint8 {
+	return 8
+}
+
+// WinnerAmount is the number of SPRs that receive a payout
+func (s4 *S4BlockGrader) WinnerAmount() int {
+	return 25
+}
+
+// AddSPRV4 verifies and adds a s4 SPR.
+func (s4 *S4BlockGrader) AddSPRV4(entryhash []byte, extids [][]byte, content []byte, pegBalance uint64) error {
+	gspr, err := ValidateS4(entryhash, extids, s4.height, content, pegBalance)
+	if err != nil {
+		return err
+	}
+	s4.sprs = append(s4.sprs, gspr)
+	return nil
+}
+
+// Grade the SPRs. The S1 algorithm works the following way:
+//	1. Take the top 50 entries with the best proof of work
+//	2. Remove top and low's 1% band from each of the 32 assets
+//	3. Calculate the average of each of the 32 assets
+//	4. Calculate the distance of each SPR to the average, where distance is the sum of quadratic differences
```

Comment:
I don't see steps 2-6 taking place. if you mind breaking down what you did that would be much appreciated

Reply:
That is grading algorithm which were implemented from version 1.0.
No need to discuss about that grading algorithm in this PR.